Knative deployer client check always fails over missing KnativeExtensionAdapter

### Describe the bug

Using `quarkus.kubernetes.deployment-target=knative` always fails as matching adapter is missing. It's a regression as this was possible with 2.16.4.

### Expected behavior

Works.

### Actual behavior

Build fails with exception:
```bash
[ERROR] 	[error]: Build step io.quarkus.kubernetes.deployment.KnativeDeployer#checkEnvironment threw an exception: java.lang.IllegalStateException: No adapter available for type:class io.fabric8.knative.client.DefaultKnativeClient
[ERROR] 	at io.fabric8.kubernetes.client.impl.BaseClient.adapt(BaseClient.java:223)
[ERROR] 	at io.quarkus.kubernetes.deployment.KnativeDeployer.lambda$checkEnvironment$0(KnativeDeployer.java:32)
[ERROR] 	at java.base/java.util.Optional.ifPresent(Optional.java:178)
[ERROR] 	at io.quarkus.kubernetes.deployment.KnativeDeployer.checkEnvironment(KnativeDeployer.java:26)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:833)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

```

### How to Reproduce?

Steps to reproduce:
1. `oc new-project mvavrik-debug`
2. `quarkus create app knative-reproducer -x openshift,resteasy-reactive --stream=3.0`
3. `cd knative-reproducer/`
4. `mvn clean verify -Dquarkus.platform.version=999-SNAPSHOT -Dquarkus.platform.group-id=io.quarkus -Dquarkus.kubernetes.deployment-target=knative -Dquarkus.kubernetes.deploy=true -Dquarkus.kubernetes-client.trust-certs=true -Dquarkus.container-image.registry=image-registry.openshift-image-registry.svc:5000`

Optional steps:
5. I added Knative Client and it made no difference
```
<dependency>
    <groupId>io.fabric8</groupId>
    <artifactId>knative-client</artifactId>
</dependency>
```


### Output of `uname -a` or `ver`

Linux

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

OpenJDK 64-Bit Server VM Temurin-17.0.5+8

### Quarkus version or git rev

999-SNAPSHOT (commit 79099039db33b9c8acc3c949818bc0c5232780fc) and 3.0.0.Alpha5

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.6

### Additional information

_No response_

--- a/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KnativeDeployer.java
+++ b/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KnativeDeployer.java
@@ -5,7 +5,7 @@ import static io.quarkus.kubernetes.deployment.Constants.KNATIVE;
 import java.util.List;
 import java.util.Optional;
 
-import io.fabric8.knative.client.DefaultKnativeClient;
+import io.fabric8.knative.client.KnativeClient;
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.kubernetes.client.spi.KubernetesClientBuildItem;
@@ -24,8 +24,8 @@ public class KnativeDeployer {
                 return;
             }
             if (target.getEntry().getName().equals(KNATIVE)) {
-                try (DefaultKnativeClient client = kubernetesClientBuilder.buildClient().adapt(DefaultKnativeClient.class)) {
-                    if (client.isSupported()) {
+                try (KnativeClient client = kubernetesClientBuilder.buildClient().adapt(KnativeClient.class)) {
+                    if (client.hasApiGroup("knative.dev", false)) {
                         deploymentCluster.produce(new KubernetesDeploymentClusterBuildItem(KNATIVE));
                     } else {
                         throw new IllegalStateException(

